modules/git/foreachref/format.go

```diff
@@ -0,0 +1,75 @@
+package foreachref
+
+import (
+	"encoding/hex"
+	"fmt"
+	"io"
+	"strings"
+)
+
+const (
+	nullChar     = "\x00"
+	dualNullChar = "\x00\x00"
+)
+
+// Format supports specifying and parsing an output format for 'git
+// for-each-ref'. See See git-for-each-ref(1) for available fields.
+type Format struct {
```

Comment:
note: if the overall approach seems reasonable I'll add unit tests for the `Format` and `Parser` types.